exercises/react/example/Reactor.ceylon

```diff
@@ -0,0 +1,25 @@
+class Reactor<Element>() {
+  shared interface Cell {
+    shared formal Element currentValue;
+  }
+
+  shared class InputCell(Element initialValue) satisfies Cell {
+    variable Element currentValue_ = initialValue;
+    shared actual Element currentValue => currentValue_;
+    assign currentValue {
+      currentValue_ = currentValue;
+    }
+  }
+
+  shared class ComputeCell satisfies Cell {
+    variable Element currentValue_;
+    Element() newValue;
+
+    shared new single(Cell c, Element(Element) f) {
```

Comment:
here I had a hope that a call `Reactor<Integer>.ComputeCell.single` would be able to treat this `Element(Element)` as `Integer(Integer)`, but it does not currently seem this is true.